#Non Graded Challenge 8

##Introduction

Nama : Gita Pramoedya Sastri

Batch : RMT-28

Objective : program ini dilakukan untuk



##Import

In [7]:
import pandas as pd


##Data Loading

In [21]:
df = pd.read_csv('https://github.com/FTDS-learning-materials/phase-0/raw/main/src/data_ngc.csv')
df

province_state country_region        date  latitude  longitude  \
0                NaN       Malaysia  2020-03-08    2.5000   112.5000   
1                NaN       Malaysia  2020-03-10    2.5000   112.5000   
2                NaN      Singapore  2020-03-13    1.3521   103.8198   
3                NaN       Malaysia  2020-03-13    4.2105   101.9758   
4                NaN       Malaysia  2020-03-15    4.2105   101.9758   
...              ...            ...         ...       ...        ...   
11925         Pahang       Malaysia  2022-06-02    3.8126   103.3256   
11926         Pahang       Malaysia  2022-04-29    3.8126   103.3256   
11927         Pahang       Malaysia  2022-01-10    3.8126   103.3256   
11928         Pahang       Malaysia  2023-02-25    3.8126   103.3256   
11929         Pahang       Malaysia  2022-09-29    3.8126   103.3256   

                location_geom  confirmed  deaths  recovered  active  fips  \
0            POINT(112.5 2.5)       99.0     0.0       24.0     NaN   NaN   
1            POINT(112.5 2.5)      129.0     0.0       24.0     NaN   NaN   
2      POINT(103.8198 1.3521)      200.0     0.0       97.0     NaN   NaN   
3      POINT(101.9758 4.2105)      197.0     0.0       26.0     NaN   NaN   
4      POINT(101.9758 4.2105)      428.0     0.0       42.0     NaN   NaN   
...                       ...        ...     ...        ...     ...   ...   
11925  POINT(103.3256 3.8126)   171734.0   975.0        NaN     NaN   NaN   
11926  POINT(103.3256 3.8126)   170740.0   970.0        NaN     NaN   NaN   
11927  POINT(103.3256 3.8126)    97320.0   782.0        NaN     NaN   NaN   
11928  POINT(103.3256 3.8126)   182730.0  1031.0        NaN     NaN   NaN   
11929  POINT(103.3256 3.8126)   177522.0  1008.0        NaN     NaN   NaN   

       admin2      combined_key  
0         NaN               NaN  
1         NaN               NaN  
2         NaN               NaN  
3         NaN               NaN  
4         NaN               NaN  
...       ...               ...  
11925     NaN  Pahang, Malaysia  
11926     NaN  Pahang, Malaysia  
11927     NaN  Pahang, Malaysia  
11928     NaN  Pahang, Malaysia  
11929     NaN  Pahang, Malaysia  

[11930 rows x 13 columns]

##Data Exploration

In [22]:
df.head()

province_state country_region        date  latitude  longitude  \
0            NaN       Malaysia  2020-03-08    2.5000   112.5000   
1            NaN       Malaysia  2020-03-10    2.5000   112.5000   
2            NaN      Singapore  2020-03-13    1.3521   103.8198   
3            NaN       Malaysia  2020-03-13    4.2105   101.9758   
4            NaN       Malaysia  2020-03-15    4.2105   101.9758   

            location_geom  confirmed  deaths  recovered  active  fips  admin2  \
0        POINT(112.5 2.5)       99.0     0.0       24.0     NaN   NaN     NaN   
1        POINT(112.5 2.5)      129.0     0.0       24.0     NaN   NaN     NaN   
2  POINT(103.8198 1.3521)      200.0     0.0       97.0     NaN   NaN     NaN   
3  POINT(101.9758 4.2105)      197.0     0.0       26.0     NaN   NaN     NaN   
4  POINT(101.9758 4.2105)      428.0     0.0       42.0     NaN   NaN     NaN   

  combined_key  
0          NaN  
1          NaN  
2          NaN  
3          NaN  
4          NaN

- cek missing value
- cek duplicate data
- cek tipe data

##Data Cleaning

In [16]:
#cek missing value

#df.drop(['active', 'fips', 'admin2', 'combined_key'], axis=1, inplace=True)

In [23]:
#cek duplicate data

df.drop_duplicates(inplace=True)

In [24]:
#cek tipe data

df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 11930 entries, 0 to 11929
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   province_state  9078 non-null   object 
 1   country_region  11930 non-null  object 
 2   date            11930 non-null  object 
 3   latitude        11321 non-null  float64
 4   longitude       11321 non-null  float64
 5   location_geom   11321 non-null  object 
 6   confirmed       11929 non-null  float64
 7   deaths          11913 non-null  float64
 8   recovered       1640 non-null   float64
 9   active          1520 non-null   float64
 10  fips            0 non-null      float64
 11  admin2          0 non-null      float64
 12  combined_key    11793 non-null  object 
dtypes: float64(8), object(5)
memory usage: 1.3+ MB


In [25]:
df['date'] = pd.to_datetime(df['date'])

In [26]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 11930 entries, 0 to 11929
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   province_state  9078 non-null   object        
 1   country_region  11930 non-null  object        
 2   date            11930 non-null  datetime64[ns]
 3   latitude        11321 non-null  float64       
 4   longitude       11321 non-null  float64       
 5   location_geom   11321 non-null  object        
 6   confirmed       11929 non-null  float64       
 7   deaths          11913 non-null  float64       
 8   recovered       1640 non-null   float64       
 9   active          1520 non-null   float64       
 10  fips            0 non-null      float64       
 11  admin2          0 non-null      float64       
 12  combined_key    11793 non-null  object        
dtypes: datetime64[ns](1), float64(8), object(4)
memory usage: 1.3+ MB


##Data Exploration

In [27]:
# mean confirmed
df['confirmed'].mean()

569167.9953893872

Dari sini dapat kita ketahui bahwa rata-rata dalam tabel confirmed itu adalah 569167.9953893872


##Conclusion

merangkum seluruh proses yang udah dijalanin
- jumlah data berapa
- summary bagaimana
- melakukan apa saja, cleaning? exploration?, dsb..